3D Dam Break With an Obstacle Example - VOF (#685)
lethe-cfd/lethe: doc/source/examples/multiphysics/3d-dam-break/3d-dam-break.rst

===============================
3D Dam Break With an Obstacle
===============================

This example simulates a dam break experiment from the Maritime Research Institute Netherlands (MARIN) `[1] <https://www.spheric-sph.org/tests/test-02>`_.

> .. warning::

Reviewer: So nice!

===============================
3D Dam Break With an Obstacle
===============================

> This example simulates a dam break experiment from the Maritime Research Institute Netherlands (MARIN) `<[1]_>`_.

Reviewer 2: ```suggestion
This example simulates a dam break experiment from the Maritime Research Institute Netherlands (MARIN) `[1] <https://www.spheric-sph.org/tests/test-02>`_.
```

.. raw:: html

    <iframe width="560" height="315" src="https://www.youtube.com/embed/XGp7pxBQvWY" frameborder="0" allowfullscreen></iframe>

In the following figure, we compare the water height evolution at 4 the positions mentioned in the `<Description of the case_>`_ section with the experimental results obtained from MARIN (available `here <https://www.spheric-sph.org/tests/test-02>`_):

.. figure:: images/H1_to_H4_evolution.png
   :alt: In this figure, the water height evolutions are compared with the experimental results of MARIN. In the top left corner, we compare the evolution of the water height at 0.496 m away from the left side of the tank. In the top right corner, we compare the evolution of the water height at 0.992 m away from the left side of the tank. In the bottom left corner, we compare the evolution of the water height at 1.488 m away from the left side of the tank. In the bottom right corner, we compare the evolution of the water height at 1.638 m away from the left side of the tank.
   :align: center
   :name: Comparison of the water height at different position in the tank with the experimental data of MARIN

> As we can see, the general evolution of the height seems to follow the experimentation results. However, on all 4 subplots, we notice that the height is overestimated. We also notice a slight shift to the right for :math:`H2`,  :math:`H3`, and :math:`H4` evolutions. These observations may be explained by the "highly viscous air" (fluid 0) that acts as an obstacle to the free flow of the water. Additionally, fluid 1 representing the water is 1000 times more viscous than regular water. With these results, we can see that the model needs to be improved to be able to simulate with accuracy low-viscosity fluids such as air.

Reviewer 2: ```suggestion
As we can see, the simulated general evolution of the height seems to follow the experimentation results. However, on all 4 subplots, we notice that the height is overestimated. We also notice a slight shift to the right for :math:`H2`,  :math:`H3`, and :math:`H4` evolutions. These observations may be explained by the "highly viscous air" (fluid 0) that acts as an obstacle to the free flow of the water. Additionally, fluid 1 representing the water is 1000 times more viscous than regular water. With these results, we can see that the model needs to be improved to be able to accurately simulate low-viscosity fluids such as air.
```

-----------
References
-----------

> .. _[1]:

Reviewer 2: ```suggestion
```

> [1] Issa, R., & Violeau, D. (2006). Test-case 2, 3D dambreaking, Release 1.1. ERCOFTAC, SPH European Research Interest Community SIG, Électricité de France, Laboratoire National d’Hydraulique et Environnement. https://www.spheric-sph.org/tests/test-02

Reviewer 2: ```suggestion
`[1] <https://www.spheric-sph.org/tests/test-02>`_ Issa, R., & Violeau, D. (2006). Test-case 2, 3D dambreaking, Release 1.1. ERCOFTAC, SPH European Research Interest Community SIG, Électricité de France, Laboratoire National d’Hydraulique et Environnement. 
```

.. warning::
    This example displays the need for improvement of low-viscosity fluid flow simulation of the current numeric model. Further work will be done to improve this aspect of the model.

----------------------------------
Features
----------------------------------
- Solver: ``gls_navier_stokes_3d`` (Q1-Q1)
- Two phase flow handled by the Volume-of-Fluids (VOF) approach with interface sharpening
- Mesh adaptation using phase fraction
- Unsteady problem handled by an adaptive BDF1 time-stepping scheme
- The use of a python script for post-processing data

Files used in this example
~~~~~~~~~~~~~~~~~~~~~~~~~~~
- Parameter file: ``examples/multiphysics/3d-dam-break/3d-dam-break.prm``
- Geometry file: ``examples/multiphysics/3d-dam-break/tank_with_obstacle.geo``
- Python script for post-processing data: ``examples/multiphysics/3d-dam-break/3d-dam-break_postprocess.py``
- Experimental data file: ``examples/multiphysics/3d-dam-break/experimental_data.txt``

.. _Description of the case:

-------------------------
Description of the case
-------------------------

> For this example, the simulated fluids are water and air. Initially, the water is at rest on the right side of the tank (represented in dashed blue lines in the figures below). At :math:`t = 0` s, the gate is opened up instantaneously and the water starts flowing under the action of gravity, :math:`\mathbf{g} = (-9.81 \  \mathbf{j}) \frac{\text{m}}{\text{s}^2}`. The tank in which this experiment happened has the following dimensions: :math:`3.22 \times 1.00 \times 1.00` m. On all boundaries, ``slip`` conditions were applied. On the left side of the tank, a rectangular box-shaped obstacle is presented (colored in grey in the figures).

Reviewer 3: ```suggestion
For this example, the simulated fluids are water and air. Initially, the water is at rest on the right side of the tank (represented in dashed blue lines in the figures below). At :math:`t = 0` s, the gate opens instantaneously and the water starts flowing under the action of gravity, :math:`\mathbf{g} = (-9.81 \  \mathbf{j}) \frac{\text{m}}{\text{s}^2}`. The tank in which this experiment happens has the following dimensions: :math:`3.22 \times 1.00 \times 1.00` m. On all boundaries, ``slip`` conditions were applied. On the left side of the tank, a rectangular box-shaped obstacle is presented (colored in grey in the figures).
```

and the max CFL is :math:`0.5`.

.. note::
    This example uses an adaptive time-stepping method, where the time-steps are modified during the simulation to keep the maximum value of the CFL condition below the given threshold (0.5).

.. code-block:: text

    #---------------------------------------------------
    # Simulation Control
    #---------------------------------------------------

    subsection simulation control
      set method                       = bdf1
      set time end                     = 6
      set time step                    = 0.001
      set adapt                        = true
      set max cfl                      = 0.5
      set output name                  = 3d-dam-break
      set output frequency             = 5
      set output path                  = ./output/
      set adaptative time step scaling = 1.05
      set output boundaries            = true
    end

Multiphysics
~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``multiphysics`` subsection enables to turn on `(true)`
and off `(false)` the physics of interest. Here ``VOF`` is chosen.

> 

Reviewer 3: ```suggestion
Note that the fluid dynamics are solved by default.
```
